# Best Advertising Markets for e-learning

The premise for this project is that of a e-learning company looking to invest money in advertising. Mainly offering courses on web and mobile development, the company also covers domains like data science, game development, etc. 

The main goal of this project is to determine which are the best two markets in which to advertise our product.

We could organize surveys for different markets to understand which would be best for advertising. However, this would be very costly and we will first look into finding a cheaper solution. Existing surveys may be relevant to our analysis. The candidate we will explore is data from [freeCodeCamp's 2017 New Coder Survey](https://www.freecodecamp.org/). A free online learning platform that offers courses on web development, freeCodeCamp also runs a popular Medium publication and their survey received reposnses from new coders in a wide variety of fields. 

The data from this survey can be found in this [GitHub Repo](https://github.com/freeCodeCamp/2017-new-coder-survey).


In [2]:
# Import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [11]:
# Read csv file
survey = pd.read_csv('2017-fCC-New-Coders-Survey-Data.csv', dtype={17: str, 62: str})

In [13]:
survey.head()

Age  AttendedBootcamp  BootcampFinish  BootcampLoanYesNo BootcampName  \
0  27.0               0.0             NaN                NaN          NaN   
1  34.0               0.0             NaN                NaN          NaN   
2  21.0               0.0             NaN                NaN          NaN   
3  26.0               0.0             NaN                NaN          NaN   
4  20.0               0.0             NaN                NaN          NaN   

   BootcampRecommend  ChildrenNumber                 CityPopulation  \
0                NaN             NaN            more than 1 million   
1                NaN             NaN              less than 100,000   
2                NaN             NaN            more than 1 million   
3                NaN             NaN  between 100,000 and 1 million   
4                NaN             NaN  between 100,000 and 1 million   

   CodeEventConferences  CodeEventDjangoGirls  ...  YouTubeFCC  \
0                   NaN                   NaN  ...         NaN   
1                   NaN                   NaN  ...         1.0   
2                   NaN                   NaN  ...         NaN   
3                   NaN                   NaN  ...         1.0   
4                   NaN                   NaN  ...         NaN   

   YouTubeFunFunFunction  YouTubeGoogleDev  YouTubeLearnCode  \
0                    NaN               NaN               NaN   
1                    NaN               NaN               NaN   
2                    NaN               NaN               1.0   
3                    1.0               NaN               NaN   
4                    NaN               NaN               NaN   

   YouTubeLevelUpTuts  YouTubeMIT  YouTubeMozillaHacks YouTubeOther  \
0                 NaN         NaN                  NaN          NaN   
1                 NaN         NaN                  NaN          NaN   
2                 1.0         NaN                  NaN          NaN   
3                 1.0         NaN                  NaN          NaN   
4                 NaN         NaN                  NaN          NaN   

   YouTubeSimplilearn  YouTubeTheNewBoston  
0                 NaN                  NaN  
1                 NaN                  NaN  
2                 NaN                  NaN  
3                 NaN                  NaN  
4                 NaN                  NaN  

[5 rows x 136 columns]

In [16]:
survey.info(verbose=True, show_counts=True)

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 18175 entries, 0 to 18174
Data columns (total 136 columns):
 #    Column                         Non-Null Count  Dtype  
---   ------                         --------------  -----  
 0    Age                            15367 non-null  float64
 1    AttendedBootcamp               17709 non-null  float64
 2    BootcampFinish                 1069 non-null   float64
 3    BootcampLoanYesNo              1079 non-null   float64
 4    BootcampName                   949 non-null    object 
 5    BootcampRecommend              1073 non-null   float64
 6    ChildrenNumber                 2314 non-null   float64
 7    CityPopulation                 15354 non-null  object 
 8    CodeEventConferences           1609 non-null   float64
 9    CodeEventDjangoGirls           165 non-null    float64
 10   CodeEventFCC                   1708 non-null   float64
 11   CodeEventGameJam               290 non-null    float64
 12   CodeEventGirlDev              

## 1. First Look and Analysis Questions
On first evaluation, the freeCodeCamp survey dataset has 18,175 rows and 136 columns. There are no columns without `NaN` values. The dataset contains information on:
* the age of the respondents 
* whether they attended a coding bootcamp or not 
* whether they finished the bootcamp 
* whether they took a loan for the bootcamp 
* the type of online resources they used (including different online learning platforms and YouTube/self-learning) 
* current employment information, desired job title 
* and much more. 

The dataset will be useful for answering our marketing questions without having to conduct a large-scale survey of our own.

### 1.1 Analysis Questions
As mentioned above, our company creates online learning content focused mainly on web and mobile development. However, we also produce content for other fields like data science, game development, etc. We are interested in answering questions about a population of new coders. We want to understand:
* Where new coders are located.
* Where are the locations with the most new coders.
* How much money new coders are willing to spend on learning.

Before we analyze the sample data in the freeCodeCamp survey, we need to verify that these data are representative for our population of interest and it has the right categories of people for our interest. 
